### PR TITLE
feat(nvidia): ujust aimode — NGC container stack deployer

### PR DESCRIPTION
**Pull request by castrojo**
## What

Single control panel for deploying NVIDIA NGC containers as Podman Quadlet user services on Bluefin NVIDIA images. One entry point: `ujust aimode`.

## Why

Bluefin NVIDIA images already have everything needed at the host level (akmod-nvidia + CDI via `nvidia-cdi-refresh.service`). This adds the user-facing layer: a discoverable, state-aware menu for deploying the full NGC stack via containers — no RPMs, no separate installers, no extra host packages.

## How it works

**`nvidia.just`** — one `[group('AI')]` recipe

`gum filter` menu over all stacks, NIM-first, showing live status (● active / ◎ deployed / ○ new). After picking a stack: detail panel (image tag, VRAM, disk) → NGC auth gate for NIM stacks → context-aware actions depending on current state:

- _New_: Deploy and start / Deploy only
- _Deployed, stopped_: Start / Open in browser / View logs / Remove
- _Running_: Open in browser / View logs / Stop / Redeploy

**`nvidia-stacks/`** — data-driven stack catalog

Each stack is a directory with `stack.env` (bash-sourceable metadata) + Podman Quadlet templates. At deploy time `envsubst '${NGC_MONTH}'` processes the templates into `~/.config/containers/systemd/`, then `systemctl --user daemon-reload`. Adding a new stack is just a new directory — the menu discovers it automatically.

| Stack | Image | Endpoints |
|---|---|---|
| `nim-llama3` | `nvcr.io/nim/meta/llama-3.1-8b-instruct:latest` | OpenAI API :8000 · needs NGC key |
| `nim-sdxl` | `nvcr.io/nim/stabilityai/stable-diffusion-xl:latest` | Image gen API :8000 · needs NGC key |
| `pytorch-lab` | `nvcr.io/nvidia/pytorch:25.06-py3` | JupyterLab :8888 |
| `tensorflow-lab` | `nvcr.io/nvidia/tensorflow:25.06-tf2-py3` | JupyterLab :8888 |
| `rapids-ds` | `nvcr.io/nvidia/rapidsai/rapidsai-core:25.02-cuda12.8-…` | JupyterLab :8888 |
| `triton-serving` | `nvcr.io/nvidia/tritonserver:25.06-py3` | HTTP :8000 / gRPC :8001 / metrics :8002 |
| `nemo-training` | `nvcr.io/nvidia/nemo:25.06` | JupyterLab :8888 |

## Architecture notes

- **Zero RPMs.** Host provides only what Bluefin nvidia images already ship.
- All quadlets use `AddDevice=nvidia.com/gpu=all` (CDI) + `--security-opt=label=disable` per [nvidia skill docs](docs/skills/nvidia.md).
- NIM API keys stored as Podman secrets (`podman secret create ngc-api-key`), not in quadlet files or env vars.
- `ngc-month` file controls the NGC release train for all monthly-versioned containers. **Follow-up needed:** Renovate regex manager to update this + `Image=` tags in `.container` files monthly.
- RAPIDS doesn't follow the NGC monthly train — pinned to `25.02-cuda12.8-runtime-ubuntu24.04-py3.12` with a comment. **Follow-up needed:** verify this tag against the NGC catalog and add to Renovate.

## Testing

Not CI-testable without a physical NVIDIA GPU. Manual verification needed on a Bluefin nvidia image:

```bash
ujust aimode          # should show gum filter menu
# pick pytorch-lab → Deploy and start
# visit http://localhost:8888
systemctl --user status pytorch-lab
journalctl --user -u pytorch-lab -f
```

For NIM: requires a free NGC API key from https://build.nvidia.com/

## Skill update

`docs/skills/nvidia.md` updated with:
- `ujust aimode` architecture decisions (one control panel, no RPMs, CDI quadlet directives)
- Ubuntu/DGX gap analysis findings
- DGX Spark (GB10 ARM64) notes for future tracking

### CODE REVIEW

**Review by hanthor — COMMENTED** [review]
## Adversarial review — optional improvements

### [MEDIUM] RAPIDS image path mismatch (docs vs code)
`docs/skills/nvidia.md` references `nvcr.io/nvidian/rapidsai/...` but the actual quadlet (`rapids-ds.container`) uses `nvcr.io/nvidia/rapidsai/rapidsai-core:25.02-cuda12.8-runtime-ubuntu24.04-py3.12`. The docs also suggest `${NGC_MONTH}` substitution but the quadlet pins an explicit tag.

**Fix:** Update the RAPIDS row in `nvidia.md` to match the actual quadlet tag and note it is pinned, not monthly.

### [LOW] Hardcoded `gum filter --height=14`
`nvidia.just:82` — if stacks grow past 14 directories, the menu clips. At 7 stacks this is fine but fragile.

**Fix:** Compute dynamically: `$(( $(ls -d "${STACKS_DIR}"/*/ | wc -l) + 2 ))` or cap with `--height=$(( $(ls ... | wc -l) + 2 > 14 ? 14 : ...))`.

### [LOW] Missing `nvidia-ctk` binary check
`nvidia.just:18` — `nvidia-ctk cdi list | grep -q` fails with a confusing error if `nvidia-ctk` is not installed at all.

**Fix:** Add `command -v nvidia-ctk >/dev/null 2>&1 || { gum style --foreground 1 "nvidia-ctk not found. GPU container stacks cannot run."; exit 1; }` before the CDI check.

### [LOW] Port conflict between NIM and Triton (both claim host:8000)
Running both `nim-llama3` and `triton-serving` simultaneously fails because both publish host port 8000. The isolated networks prevent inter-stack collisions but same-host port collision is real.

**Fix:** Add a port-conflict check in the deploy step or document the limitation.
